core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -200,6 +200,31 @@ public static void writeUint32BE(long val, byte[] out, int offset) throws ArrayI
         writeUint32BE(val, ByteBuffer.wrap(out, offset, out.length - offset));
     }
 
+    /**
+     * Write 4 bytes to the buffer as signed 32-bit integer in big endian format.
```

Comment:
It doesn't really matter whether it is singed or unsigned. We should remove "signed" from the JavaDoc here.

Reply:
This method should work for both signed and unsigned. It's using `.putInt((int) val)` which simply extracts the lower 32-bits and writes them without modification to the `ByteBuffer`, right?